### DISCUSSION TITLE
Data loader query

### DISCUSSION THREAD

**Discussion by shrimonmuke0202** [pyg-team/pytorch_geometric#2994] [Q&A]
If apply batch_size = 512 I get  adj matrix with shape (512,46,46) but when I apply batch_size = 128 I obtain some adj matrix with shape (128,44,44) but I want all adj matrix with shape (128,46,46). How can I get this type of result?

for _,data in enumerate(train_loader):
  x , edge_index, batch = data.x,data.edge_index,data.batch
  x,_ = torch_geometric.utils.to_dense_batch(x,batch)
  adj = to_dense_adj(edge_index,batch)
  print(batch)
  print(adj.shape)
  print(data.c_size)

**Comment by rusty1s** [comment]
`to_dense_adj` provides an additional argument named `max_num_nodes` that should fix your issues.

**Reply by shrimonmuke0202** [reply]
If I fixed max_num_nodes = 43 and I want all the adjacency matrices of the same dimension then an error is shown 

index 236672 is out of bounds for dimension 0 with size 236672

**Reply by rusty1s** [reply]
Can you show me an example to reproduce? For example, the following works for me:
```python
from torch_geometric.data import DataLoader
from torch_geometric.datasets import TUDataset
from torch_geometric.utils import to_dense_batch, to_dense_adj

dataset = TUDataset('/tmp/MUTAG', name='MUTAG')
loader = DataLoader(dataset, batch_size=32)

for data in loader:
    x, _ = to_dense_batch(data.x, data.batch, max_num_nodes=40)
    adj = to_dense_adj(data.edge_index, data.batch, max_num_nodes=40)
    print(x.shape, adj.shape)

```

**Reply by shrimonmuke0202** [reply]
`from torch_geometric.data import DataLoader
from torch_geometric.datasets import TUDataset
from torch_geometric.utils import to_dense_batch, to_dense_adj

dataset = TUDataset('/tmp/MUTAG', name='MUTAG')
loader = DataLoader(dataset, batch_size=32)

for data in loader:
    x, _ = to_dense_batch(data.x, data.batch)
    adj = to_dense_adj(data.edge_index, data.batch,max_num_nodes=22)
    print(x.shape, adj.shape)`

RuntimeError: index 15488 is out of bounds for dimension 0 with size 15488 this type of error occurs

**Reply by rusty1s** [reply]
`max_num_nodes` in `to_dense_batch` and `to_dense_adj` is expected to be larger than the maximum number of nodes in the graphs. I fixed this for `to_dense_adj` in master, but it is a bit trickier to implement for `to_dense_batch`.
